functions/utils/cache.test.ts

```diff
@@ -0,0 +1,42 @@
+import CacheMock from 'browser-cache-mock'
+
+import Cache from './cache'
+
+const cacheMock = new CacheMock()
+
+beforeAll(() => {
+  const globalAny: any = global
+  globalAny.caches = {
+    open: async () => cacheMock,
+    ...cacheMock,
+  }
+})
+
+test('Should put cache properly', async () => {
+  //wait for the server to start
+  const url = 'http://127.0.0.1:3000/'
+  await fetch(new Request(url)).then((res) => res.text())
+
+  const response = await Cache.match('https://example.com')
+  expect(response).toBeUndefined()
+  const data = {
+    title: 'test',
+    image: 'testImage',
+    url: 'testUrl',
+  }
+  await Cache.put(data, 'https://example.com')
+})
+
+test('Should match cache properly', async () => {
```

Comment:
These tests should be isolated - you should reset the cache between them.

For the put test, put data in and then test that it is present.
For the match test, seed data and then test that it comes out.
You can mock match/put for the appropriate tests.